### ISSUE TITLE
List index out of range exception in mwgm_graph_tool - BootEA

### ISSUE THREAD

**Issue by manuleo** [nju-websoft/OpenEA#10]
Hello
First, congratulations for the great paper and repository!
I successfully ran RDGCN matching the numbers of the paper, but I am running into troubles with BootEA. I followed the README to run BootEA on D-W V2 (15k) with `python main_from_args.py ./args/bootea_args_15k.json D_W_15k_V1 721_5fold/fold_number`. However, I am getting an index out of range exception at the 3rd fold inside the mwgm_graph_tool function. Check the error below:
```
File ..., line 112, in mwgm_graph_tool
    matched_pairs.add(pairs[index])
IndexError: list index out of range
```
I checked that when it throws the exception index=10, while the length of pairs is 6.
I also ran BootEA on D-W V1 (15k), and got the same error on the 2nd fold. Interestingly, I re-ran the code on this fold, and at the third attempt it ran smoothly. However, the average performance is 10 points below what is reported in your recent VLDB paper (MRR around 0.52).
Any idea about what might be happening?
Thanks!

**Comment by sunzequn** [comment]
Hello,

Thanks for your interest in our work!

I have checked the logs, and it seems that I did not encounter this bug. I hereby attach two running logs (submission version) of BootEA on D-W-15K-V1 for your reference. Besides, I just finished a new running of BootEA to reproduce the results, and it worked well. I also attach the new log here. 

I guess that the bug is caused by the graph-tools.  Maybe a version conflict？My version is 2.27-1. What about yours?

Thanks!

[bootea_DBP_en_WD_en_15K_V1_721_5fold_1_20190421142810.txt](https://github.com/nju-websoft/OpenEA/files/5280202/bootea_DBP_en_WD_en_15K_V1_721_5fold_1_20190421142810.txt)
[bootea_DBP_en_WD_en_15K_V1_721_5fold_2_20190421150423.txt](https://github.com/nju-websoft/OpenEA/files/5280203/bootea_DBP_en_WD_en_15K_V1_721_5fold_2_20190421150423.txt)

[bootea_D_W_15K_V1_5fold_1_20200925.txt](https://github.com/nju-websoft/OpenEA/files/5280224/bootea_D_W_15K_V1_5fold_1_20200925.txt)

**Comment by manuleo** [comment]
Hello,
thank you so much for your help! 

It turned out that the problem was exactly the graph-tool version. I downgraded it to the 2.29 (the oldest one available for my python version, 3.7.8). Now the exception disappeared and I got similar results to yours for D_W_15K_V1. 

I guess it could be a good idea to add the graph-tool version to your requirements, just to avoid other people having the same issue, but this is up to you.

Thank you and congratulations again for your work!
